src/pages/permissions/panels/EditGroupIdentitiesPanel.tsx

```diff
@@ -0,0 +1,361 @@
+import {
+  ActionButton,
+  Button,
+  Icon,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import SidePanel from "components/SidePanel";
+import { FC, useEffect, useState } from "react";
+import { queryKeys } from "util/queryKeys";
+import usePanelParams from "util/usePanelParams";
+import ScrollableTable from "components/ScrollableTable";
+import SelectableMainTable from "components/SelectableMainTable";
+import { useSearchParams } from "react-router-dom";
+import useEditHistory from "util/useEditHistory";
+import ModifiedStatusAction from "../actions/ModifiedStatusAction";
+import { pluralize } from "util/instanceBulkActions";
+import { fetchIdentities } from "api/auth-identities";
+import { LxdGroup } from "types/permissions";
+import { getCurrentIdentitiesForGroups } from "util/permissionGroups";
+import GroupIdentitiesPanelConfirmModal from "./GroupIdentitiesPanelConfirmModal";
+import PermissionIdentitiesFilter, {
+  AUTH_METHOD,
+  PermissionIdentitiesFilterType,
+  QUERY,
+} from "../PermissionIdentitiesFilter";
+import NotificationRow from "components/NotificationRow";
+
+type IdentityEditHistory = {
+  identitiesAdded: Set<string>;
+  identitiesRemoved: Set<string>;
+};
+
+interface Props {
+  groups: LxdGroup[];
+}
+
+const EditGroupIdentitiesPanel: FC<Props> = ({ groups }) => {
+  const panelParams = usePanelParams();
+  const [searchParams] = useSearchParams();
+  const notify = useNotify();
+  const [confirming, setConfirming] = useState(false);
+
+  const {
+    data: identities = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.identities],
+    queryFn: fetchIdentities,
+  });
+
+  const {
+    desiredState,
+    save: saveToPanelHistory,
+    undo: undoIdentitiesChange,
+  } = useEditHistory<IdentityEditHistory>({
+    initialState: {
+      identitiesAdded: new Set(),
+      identitiesRemoved: new Set(),
+    },
+  });
+
+  if (error) {
+    notify.failure("Loading panel details failed", error);
+  }
+
+  // in case if user refresh the browser while the panel is open
+  useEffect(() => {
+    if (!groups.length) {
+      panelParams.clear();
+      return;
+    }
+  }, [groups]);
+
+  const nonTlsIdentities = identities.filter(
+    (identity) => identity.authentication_method !== "tls",
+  );
+
+  const calculateIdentityStatus = () => {
+    const { identityIdsInAllGroups, identityIdsInSomeGroups } =
+      getCurrentIdentitiesForGroups(groups, nonTlsIdentities);
+
+    const identitiesAllocatedToAll = new Set<string>();
+    const identitiesAllocatedToSome = new Set<string>();
```

Comment:
To avoid confusion with the `identityIdsInAllGroups` and `identityIdsInSomeGroups`,  shall we call those `selectedIdentities` and `indeterminedIdentities`?

Same for the other components that use the FooAllocatedTo* pattern.

Reply:
nice, now that `selectedIdentities` is no longer an input prop, this certainly works, it's been challenging to get to an agreeable naming convention for these really, thanks! The same changes have been applied to `EditIdentityGroupsPanel` too.